### PR TITLE
[PeerDAS] implement DataColumnSidecarsByRootReq and fix related bugs

### PR DESCRIPTION
**Pull request by 0x00101010**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**
implement DataColumnSidecarsByRootReq and fix related bugs

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: beacon-chain/p2p/types/types.go [hunk 1]
```diff
@@ -207,7 +208,94 @@ func (s BlobSidecarsByRootReq) Len() int {
...
+
+// DataColumnIdentifier is a fixed size value, so we can compute its fixed size at start time (see init below)
+var dataColumnIdSize int
+
+// UnmarshalSSZ implements ssz.Unmarshaler. It unmarshals the provided bytes buffer into the DataColumnSidecarsByRootReq value.
+func (d *DataColumnSidecarsByRootReq) UnmarshalSSZ(buf []byte) error {
+	bufLen := len(buf)
+	maxLen := int(params.BeaconConfig().MaxRequestDataColumnSidecars) * dataColumnIdSize
+	if bufLen > maxLen {
+		return errors.Errorf("expected buffer with length of up to %d but received length %d", maxLen, bufLen)
```
Comment (nalepae) [inline beacon-chain/p2p/types/types.go:229]: Could you add a test for this?

File: beacon-chain/p2p/types/types.go [hunk 2]
```diff
@@ -207,7 +208,94 @@ func (s BlobSidecarsByRootReq) Len() int {
...
+			return nil, err
+		}
+		copy(buf[i*dataColumnIdSize:(i+1)*dataColumnIdSize], bytes)
+	}
+
+	return buf, nil
+}
+
+// MarshalSSZTo implements ssz.Marshaler. It appends the serialized DataColumnSidecarsByRootReq value to the provided byte slice.
+func (d *DataColumnSidecarsByRootReq) MarshalSSZTo(dst []byte) ([]byte, error) {
```
Comment (nalepae) [inline beacon-chain/p2p/types/types.go:262]: Could you add a test for this?

File: beacon-chain/p2p/types/types.go [hunk 3]
```diff
@@ -207,7 +208,94 @@ func (s BlobSidecarsByRootReq) Len() int {
...
+	return append(dst, mobj...), nil
+}
+
+// SizeSSZ implements ssz.Marshaler. It returns the size of the serialized representation.
+func (d *DataColumnSidecarsByRootReq) SizeSSZ() int {
+	return len(*d) * dataColumnIdSize
+}
+
+// Len implements sort.Interface. It returns the number of elements in the collection.
+func (d DataColumnSidecarsByRootReq) Len() int {
```
Comment (nalepae) [inline beacon-chain/p2p/types/types.go:276]: Nit: Could you add a test for this?

File: beacon-chain/p2p/types/types.go (hunk 2 above)
Comment (0x00101010) [reply beacon-chain/p2p/types/types.go:262]: Added

File: beacon-chain/p2p/types/types.go (hunk 3 above)
Comment (0x00101010) [reply beacon-chain/p2p/types/types.go:276]: Added

File: beacon-chain/p2p/types/types.go (hunk 1 above)
Comment (0x00101010) [reply beacon-chain/p2p/types/types.go:229]: Added

**Comment by nalepae** [comment]
Thanks!

**Pull request merged**